sdk/nodejs/policy/policy.ts

```diff
@@ -102,8 +102,67 @@ export interface Policy {
      * proper permissions.
      */
     enforcementLevel?: EnforcementLevel;
+
+    /**
+     * Configuration schema.
+     */
+    config?: PolicyConfigSchema;
+}
+
+export interface PolicyConfigSchema {
+    properties: Record<string, PolicyConfigPropertySchema>;
+    required?: string[];
 }
 
+export type PolicyConfigPropertySchemaDefinition = PolicyConfigPropertySchema | boolean;
+export interface PolicyConfigPropertySchema {
+    type?: PolicyConfigPropertySchemaTypeName | PolicyConfigPropertySchemaTypeName[];
+    enum?: PolicyConfigPropertySchemaType[];
+    const?: PolicyConfigPropertySchemaType;
+
+    multipleOf?: number;
+    maximum?: number;
+    exclusiveMaximum?: number;
+    minimum?: number;
+    exclusiveMinimum?: number;
+
+    maxLength?: number;
+    minLength?: number;
+    pattern?: string;
+
+    items?: PolicyConfigPropertySchemaDefinition | PolicyConfigPropertySchemaDefinition[];
+    additionalItems?: PolicyConfigPropertySchemaDefinition;
+    maxItems?: number;
+    minItems?: number;
+    uniqueItems?: boolean;
+    contains?: PolicyConfigPropertySchema;
+
+    maxProperties?: number;
+    minProperties?: number;
+    required?: string[];
+    properties?: {
+        [key: string]: PolicyConfigPropertySchemaDefinition;
+    };
+    patternProperties?: {
+        [key: string]: PolicyConfigPropertySchemaDefinition;
+    };
+    additionalProperties?: PolicyConfigPropertySchemaDefinition;
+    dependencies?: {
+        [key: string]: PolicyConfigPropertySchemaDefinition | string[];
+    };
+    propertyNames?: PolicyConfigPropertySchemaDefinition;
+
+    description?: string;
+    default?: PolicyConfigPropertySchemaType;
+    readOnly?: boolean;
+    writeOnly?: boolean;
+}
```

Comment:
It's not clear to me we want to expose all this, at least not initially. Should we just expose a minimal useful subset and then see if there's feedback before exposing more? If so, what would the subset look like? I just added most of it from [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a9a4d049259d3748c2d1440238489cbccf3cd395/types/json-schema/index.d.ts#L545) as a stand-in.

Reply:
I am not sure `readOnly` and `writeOnly` make sense here. I think my preference would be to start with a smaller batch of well tested ones and then grow it with time. I think for starters we should support: type, enum, all the num validation ones, all the string validation ones, and all the list validation ones, all the property validation ones... and is i begin to compile this list I ask may be we just support them all 🤔

Reply:
Here's what I've done:

- Removed `readOnly` and `writeOnly`.
- Added `format`.
- Kept the rest that I had.

Here's all of the properties (json schema 7) that I've left off for now:

```typescript
    $id?: string;
    $ref?: string;
    $schema?: JSONSchema7Version;
    $comment?: string;

    if?: JSONSchema7Definition;
    then?: JSONSchema7Definition;
    else?: JSONSchema7Definition;

    allOf?: JSONSchema7Definition[];
    anyOf?: JSONSchema7Definition[];
    oneOf?: JSONSchema7Definition[];
    not?: JSONSchema7Definition;

    contentMediaType?: string;
    contentEncoding?: string;

    definitions?: {
        [key: string]: JSONSchema7Definition;
    };

    title?: string;
    readOnly?: boolean;
    writeOnly?: boolean;
    examples?: JSONSchema7Type;
```

I also decided to move most of the schema types out of the main module, to avoid the unnecessary noise when importing via `import * as policy from "@pulumi/policy"`.

And I added some initial doc comments, though, as a follow-up, I'd like to add doc comments to all of these.